addon/components/rdfa/rdfa-editor.ts

```diff
@@ -99,48 +114,43 @@ export default class RdfaEditor extends Component {
    *
    * @private
    */
-  get hasActiveHints() {
-    return this.hintsRegistry?.activeHints?.length > 0;
+  get hasActiveHints(): boolean {
+    return this.hintsRegistry?.activeHints
+      ? this.hintsRegistry?.activeHints?.length > 0
+      : false;
   }
 
   /**
    * @property hasSuggestedHints
    */
-  get hasSuggestedHints() {
+  get hasSuggestedHints(): boolean {
     return this.suggestedHints.length > 0;
   }
 
   /**
    * @property rootModelNode
    */
-  get rootModelNode() {
+  get rootModelNode(): ModelNode {
     return this.editor.rootModelNode;
   }
-  /**
-   * Contains extra handlers for input events on the editor.
-   *
-   * @property handlers
-   * @type Ember.A
-   *
-   * @private
-   */
-  handlers = null;
 
   /**
    * editor controller
-   *
    */
-  @tracked editor;
+  @tracked editor?: PernetRawEditor;
 
-  init() {
-    super.init(...arguments);
-    const userLocale = ( navigator.language || navigator.languages[0] );
+  constructor(owner: unknown, args: RdfaEditorArgs) {
+    super(owner, args);
+    const userLocale = (navigator.language || navigator.languages[0]);
     this.intl.setLocale([userLocale, 'nl-BE']);
-    this.set('handlers', A());
+    if (this.args.profile) {
+      this.profile = this.args.profile;
+    }
   }
 
-  didUpdateAttrs() {
-    if (this.profile != this.eventProcessor.profile) {
+  @action
+  updateProfile() {
+    if (this.eventProcessor && this.profile != this.eventProcessor.profile) {
```

Comment:
might be worth going through and making sure we use triple equals everywhere (kinda surprised and slightly alarmed this doesn't trigger the linter)

Reply:
that would be a lot of linter errors:
```
nielsv@smallbox ~/Code/say-editor/ember-rdfa-editor $ grep -r ' == ' addon/ | wc -l
164
nielsv@smallbox ~/Code/say-editor/ember-rdfa-editor $ grep -r ' != ' addon/ | wc -l
27
```